### PR TITLE
Chore: remove dm

### PR DESCRIPTION
**Pull request by Somefive**
### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: cmd/core/app/server.go [hunk 1]
```diff
@@ -177,12 +176,6 @@ func run(ctx context.Context, s *options.CoreOptions) error {
 		return err
 	}
 
-	dm, err := discoverymapper.New(mgr.GetConfig())
-	if err != nil {
-		klog.ErrorS(err, "Failed to create CRD discovery client")
-		return err
-	}
-	s.ControllerArgs.DiscoveryMapper = dm
 	pd, err := packages.NewPackageDiscover(mgr.GetConfig())
```
Comment (FogDong) [inline cmd/core/app/server.go:179]: Maybe we can deprecate `packageDiscover` as well.

**Review by FogDong — APPROVED** [review]

File: cmd/core/app/server.go (hunk 1 above)
Comment (Somefive) [reply cmd/core/app/server.go:179]: Yes. I would like to do so as well. Maybe in next PR.

File: pkg/appfile/parser.go [hunk 2]
```diff
@@ -773,15 +769,14 @@ func (p *Parser) ValidateComponentNames(app *v1beta1.Application) (int, error) {
...
-	name string) (*v1beta1.ScopeDefinition, metav1.GroupVersionKind, error) {
+func GetScopeDefAndGVK(ctx context.Context, cli client.Client, name string) (*v1beta1.ScopeDefinition, metav1.GroupVersionKind, error) {
 	var gvk metav1.GroupVersionKind
 	sd := new(v1beta1.ScopeDefinition)
 	err := util.GetDefinition(ctx, cli, sd, name)
 	if err != nil {
 		return nil, gvk, err
 	}
-	gvk, err = util.GetGVKFromDefinition(dm, sd.Spec.Reference)
+	gvk, err = util.GetGVKFromDefinition(cli.RESTMapper(), sd.Spec.Reference)
```
Comment (wonderflow) [inline pkg/appfile/parser.go:779]: does the restmapper equal to discovermapper here?

Comment (Somefive) [reply pkg/appfile/parser.go:779]: Yes

**Review by chivalryq — APPROVED** [review]
LGTM

**Review by wonderflow — APPROVED** [review]

**Pull request merged**